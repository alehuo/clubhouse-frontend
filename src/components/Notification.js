import React from "react";
import { Alert } from "react-bootstrap";
const Notification = ({ text, type }) => {
  return (
    <Alert bsStyle={type}>
      <p>{text}</p>
    </Alert>
  );
};

export default Notification;
