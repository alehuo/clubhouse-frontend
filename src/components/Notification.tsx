import React from "react";
import { Alert } from "react-bootstrap";

interface Props {
  text: string;
  type: "success" | "warning" | "danger" | "info";
}

const Notification: React.FC<Props> = ({ text, type }) => (
  <Alert variant={type}>
    <p>{text}</p>
  </Alert>
);

export default Notification;
