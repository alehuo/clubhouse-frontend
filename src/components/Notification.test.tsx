import { shallow } from "enzyme";
import React from "react";
import Notification from "./Notification";

it("Notification matches snapshot", () => {
  const cmpnt = shallow(<Notification text="Hello World" type="error" />);
  expect(cmpnt).toMatchSnapshot();
});
