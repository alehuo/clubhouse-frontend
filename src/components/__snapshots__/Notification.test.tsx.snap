// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notification
    text="Hello World"
    type="error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "alert",
      "bsStyle": "error",
      "children": <p>
        Hello World
      </p>,
      "closeLabel": "Close alert",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World",
      },
      "ref": null,
      "rendered": "Hello World",
      "type": "p",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "alert",
        "bsStyle": "error",
        "children": <p>
          Hello World
        </p>,
        "closeLabel": "Close alert",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World",
        },
        "ref": null,
        "rendered": "Hello World",
        "type": "p",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
